plugin.video.mediathekview/resources/lib/downloader.py

```diff
@@ -0,0 +1,361 @@
+# -*- coding: utf-8 -*-
+"""
+Download handler module
+
+Copyright (c) 2018, Leo Moll
+Licensed under MIT License
+"""
+
+# pylint: disable=import-error
+# pylint: disable=mixed-indentation, bad-whitespace, bad-continuation, missing-docstring
+
+# -- Imports ------------------------------------------------
+from __future__ import unicode_literals
+
+import os
+import re
+
+from contextlib import closing
+
+import xbmc
+import xbmcvfs
+
+import resources.lib.mvutils as mvutils
+
+from resources.lib.kodi.KodiUI import KodiBGDialog
+from resources.lib.filmui import FilmUI
+from resources.lib.ttml2srt import ttml2srt
+
+# -- Classes ------------------------------------------------
+class Downloader( object ):
+	def __init__( self, plugin ):
+		self.plugin		= plugin
+		self.database	= plugin.database
+		self.settings	= plugin.settings
+		self.notifier	= plugin.notifier
+
+	def play_movie_with_subs( self, filmid, external ):
+		film = self.database.RetrieveFilmInfo( filmid )
+		if film is None:
+			self.notifier.ShowError( 30990, self.plugin.language( 30991 ) )
+			return
+		ttmname = self.plugin.datapath + '/subtitle.ttml'
```

Comment:
it would be nice to use the os module to join paths

Reply:
You are right. Will fix it.